lua/ttt2/libraries/vguihandler.lua

```diff
@@ -0,0 +1,167 @@
+---
+-- A vgui handler
+-- @author Mineotopia
+
+AddCSLuaFile()
+
+-- the rest of the vguihandler library is client only
+if SERVER then return end
+
+local table = table
+
+vguihandler = vguihandler or {
+	frames = {},
+	callback = {
+		frame = {},
+		hidden = {}
+	}
+}
+
+-- Call this function to create a new frame. Using this instead of setting the
+-- frame up manually has the benefit that all basic settings are already set.
+-- Additionally the frame is added to a managed list that is used to handle
+-- hiding, closing and tracking menues.
+-- @param number w The width of the panel
+-- @param number h The height of the panel
+-- @param string title The title of the panel
+-- @return @{Panel} The created/cleared DFrameTTT2 object
+function vguihandler.GenerateFrame(w, h, title)
+	local frame = vgui.Create("DFrameTTT2")
+
+	frame:SetSize(w, h)
+	frame:Center()
+	frame:SetTitle(title)
+
+	local OriginalClose = frame.Close
+
+	frame.Close = function(slf)
+		if slf:GetDeleteOnClose() then
+			table.RemoveByValue(vguihandler.frames, frame)
+		end
+
+		OriginalClose(slf)
+	end
+
+	vguihandler.frames[#vguihandler.frames + 1] = frame
+
+	return frame
+end
+
+---
+-- Hides all registered and unhidden frames.
+-- @return table Returns a table of the frames that are now hidden
+-- @realm client
+function vguihandler.HideUnhiddenFrames()
+	local frames = vguihandler.frames
+	local hiddenFrames = {}
+
+	for i = 1, #frames do
```

Comment:
`#frames` should be cached. Below the same

Reply:
In none of these functions `#frames` is used more than once